### DISCUSSION TITLE
FundMe.sol gas requirement is infinite

### DISCUSSION THREAD

**Discussion by guillermo-palou** [smartcontractkit/full-blockchain-solidity-course-py#310] [Q&A]
I am getting the following error on Lesson 3:

Gas requirement of function FundMe.getConversioRate is infinite: If the gas requirement of a function is higher than the block gas limit, it cannot be executed.

Here is the getConversionRate, I don't see any loops or anything

function getConversionRate(uint256 ethAmount) public view returns (uint256){
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        return ethAmountInUsd;
    }

**Comment by cromewar** [comment]
Hello @guillermo-palou, make sure your `getPrice()` function looks like this:

    
  ```Solidity
  function getPrice() public view returns(uint256){
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);
        (,int256 answer,,,) = priceFeed.latestRoundData();
         return uint256(answer * 10000000000);
    }
```

Also don't forget to setup the correct amount of gas on REMIX

![image](https://user-images.githubusercontent.com/20199180/140087078-94c7d819-74aa-4d6c-9238-dee458dd55b3.png)
